source/Octopus.Tentacle/Kubernetes/KubernetesAgentToolsImageVersionMetadataProvider.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using NuGet.Versioning;
+using Octopus.Diagnostics;
+
+namespace Octopus.Tentacle.Kubernetes
+{
+    public interface IToolsImageVersionMetadataProvider
+    {
+        Task<KubernetesAgentToolsImageVersionMetadata?> TryGetVersionMetadata();
+    }
+
+    public class KubernetesAgentToolsImageVersionMetadataProvider : IToolsImageVersionMetadataProvider
+    {
+        readonly ISystemLog log;
+
+        public KubernetesAgentToolsImageVersionMetadataProvider(ISystemLog log)
+        {
+            this.log = log;
+        }
+        
+        public async Task<KubernetesAgentToolsImageVersionMetadata?> TryGetVersionMetadata()
+        {
+            using var httpClient = new HttpClient();
+            try
+            {
+                var response = await httpClient.GetAsync("https://oc.to/kubernetes-agent-tools-image-metadata");
```

Comment:
Using a short URL allows us to avoid updating Tentacle if the location of the metadata file changes.